### PR TITLE
[SPARK-7272] [MLLIB] User guide for PMML model export

### PR DESCRIPTION
**Pull request by selvinsource**
https://issues.apache.org/jira/browse/SPARK-7272

### CODE REVIEW

**Comment by selvinsource** [comment]
The contribution is my original work and I license the work to the project under the project's open source license.

**Comment by mengxr** [comment]
@selvinsource Could you update the PR title use this format: `[SPARK-7272] [MLLIB]....`, and make the description more accurate? For example, this should be `[SPARK-7272] [MLLIB] User guide for PMML model export`. Thanks!

File: docs/mllib-pmml-model-export.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+val data = sc.textFile("data/mllib/kmeans_data.txt")
+val parsedData = data.map(s => Vectors.dense(s.split(' ').map(_.toDouble))).cache()
+
+// Cluster the data into two classes using KMeans
+val numClusters = 2
+val numIterations = 20
+val clusters = KMeans.train(parsedData, numClusters, numIterations)
+
+// Export to PMML
+println("PMML export = " + clusters.toPMML)
```
Comment (mengxr) [inline docs/mllib-pmml-model-export.md]: Remove `"PMML export = " + ` or change it to `PMML Model:\n". It is hard to think of a use case that mixes some strings with the XML content in the same line.

**Comment by mengxr** [comment]
LGTM. Merged into master and branch-1.4. Thanks! Btw, we may need to reorganize the layout. The current TOC of the user guide is really long.

**Comment by selvinsource** [comment]
Great!

**Pull request closed without merging**